### PR TITLE
Clean staging

### PR DESCRIPTION
**Pull request by arjun4084346**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-XXX

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java [hunk 1]
```diff
@@ -439,6 +450,7 @@ public void launchJob(JobListener jobListener)
       TimingEvent launchJobTimer = this.eventSubmitter.getTimingEvent(TimingEvent.LauncherTimings.FULL_JOB_EXECUTION);
 
       try (Closer closer = Closer.create()) {
+        addShutdownHook(closer);
```
Comment (ZihanLi58) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java:453]: what's the shutdown hook doing here? Seems like the closer is created in the try clause which should be automatically closed? Am I missing any thing here?

**Review by ZihanLi58 — CHANGES_REQUESTED** [review]
I don't see where the clean of staging folder happens, can you explain that a little? Also the test fails

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java (hunk 1 above)
Comment (arjun4084346) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java:453]: the closer that is created in `try` is not closed when there is a kill. `finally` blocks does not run when the jvm is abruptly stopping. Adding a shutdown hook makes sure that close is called in those cases.

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java [hunk 2]
```diff
@@ -679,6 +694,7 @@ public void apply(JobListener jobListener, JobContext jobContext)
             }
           }
         }
+        jobStoppedGracefully = true;
```
Comment (ZihanLi58) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java:697]: We should set this in final block?

Comment (arjun4084346) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java:697]: Yes, this means that the finally got to execute in a normal way, so shutdown hook should not double try to close again.
Did that answer your comment?

**Pull request closed without merging**